src/cartesian/Area.tsx

```diff
@@ -666,7 +666,7 @@ export class Area extends PureComponent<Props, State> {
             </defs>
           ) : null}
           {!hasSinglePoint ? this.renderArea(needClip, clipPathId) : null}
-          {(dot || hasSinglePoint) && this.renderDots(needClip, clipDot, clipPathId)}
+          {(dot || (hasSinglePoint && dot !== false)) && this.renderDots(needClip, clipDot, clipPathId)}
```

Comment:
so I think if we were going to do this we would just remove the `hasSinglePoint` check altogether. Then when dot is falsy the dots wouldn't render. Adding an additional condition only checking for false is confusing.

Reply:
Since the  `dot` prop is optional, it leaves some field for people using default to have the default behavior.

My thoughts are that the dots should not be there at all when not set to `true` but trying to fit with the will to not change the default behavior.

Reply:
its defaulted to false here https://github.com/recharts/recharts/blob/3.x/src/cartesian/Area.tsx#L171

so this would still be the case as the new default behavior unless the default itself changed.

Reply:
Thanks for the feedback. So the only way to do this is to just remove `hasSinglePoint` condition.